webdriver/tests/classic/execute_script/node.py

```diff
@@ -57,6 +68,30 @@ def test_stale_element(session, get_test_page, as_frame):
     assert_error(result, "stale element reference")
 
 
+@pytest.mark.parametrize("expression, expected_type", [
+    (""" document.querySelector("svg").attributes[0] """, "attribute"),
+    (""" document.querySelector("div#text-node").childNodes[1] """, "text"),
+    (""" document.implementation.createDocument("", "root", null).createCDATASection("foo") """, "cdata"),
+    (""" document.createProcessingInstruction("xml-stylesheet", "href='foo.css'") """, "processing_instruction"),
+    (""" document.querySelector("div#comment").childNodes[0] """, "comment"),
+    (""" document""", "document"),
+    (""" document.doctype""", "doctype"),
+], ids=["attribute", "text", "cdata", "processing_instruction", "comment", "document", "doctype"])
+def test_node_type(session, inline, expression, expected_type):
+    session.url = inline(PAGE_DATA)
+
+    response = execute_script(
+        session,
+        f"const result = {expression}; return {{'result': result, 'type': result.nodeType}}",
+    )
+    result = assert_success(response)
+    if expected_type == "document":
+        assert 'location' in result['result']
+    else:
+        assert result['result'] == {}
+    assert result["type"] == NODE_TYPE[expected_type]
```

Comment:
I would move this before the other assertions which would then indicate that we serialized a wrong node. Otherwise it might be harder to understand why eg `location` is not included.